calculator_test.go

```diff
@@ -2,23 +2,416 @@ package calculator_test
 
 import (
 	"calculator"
+	"math/rand"
 	"testing"
+	"time"
 )
 
+// Note Fatal|FatalF are also useful methods,
+// to stop further execution within the same test.
+
+// Express test-case inputs and desired outputs
+type testCase struct {
```

Comment:
Since test case struct types tend to be specific to the test they're used in, we often don't bother to give them a name. Instead, we define an _anonymous struct type_:

```go
tcs := []struct{
	a, b, want float64
	...
}{
	{ a: 2, b: 2, ... },
}
```